cloudvolume/volumecutout.py

```diff
@@ -0,0 +1,92 @@
+import os
+
+from six.moves import range
+import numpy as np
+from PIL import Image
+from tqdm import tqdm
+
+from .lib import mkdir
+
+class VolumeCutout(np.ndarray):
+
+  def __new__(cls, buf, dataset_name, layer, mip, layer_type, bounds, *args, **kwargs):
+    return super(VolumeCutout, cls).__new__(cls, shape=buf.shape, buffer=np.ascontiguousarray(buf), dtype=buf.dtype)
+
+  def __init__(self, buf, dataset_name, layer, mip, layer_type, bounds, *args, **kwargs):
+    super(VolumeCutout, self).__init__()
+    
+    self.dataset_name = dataset_name
+    self.layer = layer
+    self.mip = mip
+    self.layer_type = layer_type
+    self.bounds = bounds
+
+  @classmethod
+  def from_volume(cls, volume, buf, bounds):
+    return VolumeCutout(
+      buf=buf,
+      dataset_name=volume.dataset_name,
+      layer=volume.layer,
+      mip=volume.mip,
+      layer_type=volume.layer_type,
+      bounds=bounds,
+    )
+
+  @property
+  def num_channels(self):
+    return self.shape[3]
+
+  def save_images(self, axis='z', channel=None, directory=None, image_format='PNG'):
+
+    if directory is None:
+      directory = os.path.join('./saved_images', self.dataset_name, self.layer, str(self.mip), self.bounds.to_filename())
+    
+    mkdir(directory)
+
+    print("Saving to {}".format(directory))
+
+    indexmap = {
```

Comment:
IntEnum is a better fitting?
https://docs.python.org/3/library/enum.html#intenum

Reply:
Possibly, but I this is simpler and works in this small situation.